src/order/hom/basic.lean

```diff
@@ -683,3 +687,29 @@ theorem order_iso.is_complemented_iff :
 
 end bounded_order
 end lattice_isos
+
+section boolean_algebra
+variables (α) [boolean_algebra.core α]
+
+/-- Taking complements as an order homomorphism to the order dual. -/
+@[simps]
+def order_hom.compl : α →o order_dual α :=
+{ to_fun := order_dual.to_dual ∘ compl,
+  monotone' := λ x y, compl_le_compl }
+
+/-- Taking complements as an order isomorphism to the order dual. -/
+@[simps]
+def order_hom.compl_iso : α ≃o order_dual α :=
```

Comment:
Should this be in the `order_hom` namespace? And should there be a simp-lemma that says if you coerce this to an order_hom then you end up with `order_hom.compl`?

Reply:
Whoops, you're right
```suggestion
def order_iso.compl : α ≃o order_dual α :=
```
However, `order_hom.compl` doesn't exist and I don't think there's a point for it, as the required instances aren't weaker.

Reply:
Oh wait, it's been written. Well, I think we don't need it then.

Reply:
My goal was to do it in a similar way to e.g., `order_hom.pi` and `order_hom.pi_iso`, but I'm happy to change it back.